pubsub/natspubsub/nats_streaming.go

```diff
@@ -0,0 +1,277 @@
+package natspubsub
+
+import (
+	"context"
+	"errors"
+	"sync"
+
+	"github.com/nats-io/nats.go"
+	"github.com/nats-io/stan.go"
+
+	"gocloud.dev/gcerrors"
+	"gocloud.dev/pubsub"
+	"gocloud.dev/pubsub/driver"
+)
+
```

Comment:
There's no URL openers defined here...?

Also, a fair amount of the code is already duplicated with `nats.go` (and adding the URL opener will increase it). It might be simpler to use the same file, the same struct, and do conditionals where needed?

E.g.,

```
type topic struct {
        // Exactly one of nc, snc will be non-nil.
	nc   *nats.Conn
        snc stan.Conn
	subj string
}
```

The URL opener could have a URL param to indicate to use streaming, or a separate environment variable (`NATS_STREAMING_SERVER_URL`).